Added Centrality Algorithms Example Jupyter Notebook

kedarghule opened the pull request:
This PR adds an example jupyter notebook for centrality algorithms on the airline travel reachability network dataset. The notebook explores eigenvector centrality, betweenness centrality & degree centrality.

This is in reference to my comment in the issue: https://github.com/neo4j/graph-data-science-client/issues/254

Thank you :)

kedarghule commented:
Umm. I don't know why one check is not running :/

Mats-SX commented:
@kedarghule The check not running is normal. For security reasons we do not run PR builds from non-members. There have been several attacks made on our public repositories to attempt to expose secrets from our build system.

I apologise for the delayed review. We are in the process of releasing GDS `2.4.0` which is taking up most of my time lately. We are grateful for your contribution, and you are not forgotten :)

All the best
Mats

kedarghule commented:
> @kedarghule The check not running is normal. For security reasons we do not run PR builds from non-members. There have been several attacks made on our public repositories to attempt to expose secrets from our build system.
> 
> I apologise for the delayed review. We are in the process of releasing GDS `2.4.0` which is taking up most of my time lately. We are grateful for your contribution, and you are not forgotten :)
> 
> All the best Mats

Alright @Mats-SX  and sure, no problem :)

Mats-SX reviewed (approved):
Thanks a lot for this contribution! It's a great notebook. I only have a few comments.

I will push a commit that clears output cells and applies our code style rules to the Python code. For reference, we have two scripts in the `scripts` directory: `checkstyle` and `makestyle` that can be used (just run them as `scripts/checkstyle` without arguments) to manage this in future contributions.

I will also create a sibling PR to allow our CI to run.

kedarghule commented on examples/centrality-algorithms.ipynb:
```diff
@@ -0,0 +1,991 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we load this data (nodes and edges) into a Graph Database and a GDS graph."
```
Alright. Done :)

kedarghule commented on examples/centrality-algorithms.ipynb:
```diff
@@ -0,0 +1,991 @@
+    "\n",
+    "# We can use convenience methods on `G` to check if the projection looks correct\n",
+    "print(f\"Graph '{G.name()}' node count: {G.node_count()}\")\n",
+    "print(f\"Graph '{G.name()}' node labels: {G.node_labels()}\")"
```
Done!

kedarghule commented on examples/centrality-algorithms.ipynb:
```diff
@@ -0,0 +1,991 @@
+    "Austin R. Benson, David F. Gleich, and Jure Leskovec. \"Higher-order Organization of Complex Networks.\" Science, 353.6295 (2016): 163–166.\n",
+    "\n",
+    "- Notebook contributed by [Kedar Ghule](https://github.com/kedarghule)"
+   ]
```
Done!

kedarghule commented:
@Mats-SX I have addressed your comments for the PR and made a new commit. Please let me know if anything else is needed or if I missed anything.

Also the latest commit won't run a check again :/

Mats-SX commented on examples/centrality-algorithms.ipynb:
```diff
@@ -0,0 +1,991 @@
+       " 'p95': 0.08167456835508347,\n",
+       " 'p75': 0.05822562426328659}"
+      ]
+     },
```
Yeah I noticed that our `makestyle` script does this automatically :)

Mats-SX commented on examples/centrality-algorithms.ipynb:
```diff
@@ -438,13 +462,29 @@
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.10.11"
```
This stuff was added back in, which is normal when the notebook is run. I will clear it out again.

Mats-SX commented:
@kedarghule Thank you! I've kicked off another CI build.
